bin/p2-install-hook/main.go

```diff
@@ -49,11 +50,19 @@ func main() {
 
 	if *isPreparerManifest {
 		config := manifest.GetConfig()
-		hooksManifest, ok := config["hooks_manifest"]
-		if ok {
-			manifest = hooksManifest.(p2manifest.Manifest)
-		} else {
-			fmt.Printf("Error to read hooks manifest from P2-Preparer manifest: %v", manifest)
+		preparerConfigBytes, err := yaml.Marshal(config)
```

Comment:
Following further inspection of `manifest.GetConfig()`, it looks like it already does the `Marshal` and `Unmarshal` steps that you are attempting here. https://github.com/square/p2/blob/master/pkg/manifest/manifest.go#L185-L208 The only difference between the two is that you are using `preparer.UnmarshalConfig()` instead of `yaml.Unmarshal()`. As best I can tell, the reason you want to explicitly use the `preparer.UnmarshalConfig()` over just a `yaml.Unmarshal()` is because it adds the hostname, consul address, hooks directory, and pod root if those values are missing from the manifest. https://github.com/square/p2/blob/master/pkg/preparer/setup.go#L265-L291

As you specifically want the hooks data and you do not want that info to be missing, and you also don't want the configuration to change unexpectedly, should we be using `manifest.Marshal(config, err)` instead of just `yaml.Marshal(config)`? https://github.com/square/p2/blob/master/pkg/manifest/manifest.go#L400-L413

Reply:
There is no `manifest.Marshal(config, err)` there is just manifest.Marshal() ([]byte, err) which cannot be used in this case, as manifest is manifest, we want to marshal `config`.

The reason why we need to marshal and unmarshal twice is we need to convert `config` to `preparerConfig` which is unique to have hooksmanifest. 

The only way to avoid this is to add `manfiest.getPreparerConfig()` method, which is not optimal as normal manifest does not preparer config, and this is not frequently used.